packages/tc-api-rest-handlers/lib/events/__tests__/events-integration.spec.js

```diff
@@ -226,6 +225,31 @@ describe('Rest events module integration', () => {
 			]);
 			expectUpdateEvent(childType, childCode, ['isChildOf']);
 		});
+
+		it('should not send extra UPDATE events when removing rich relationship property that was already non-existent', async () => {
+			const [main, child] = await Promise.all([
+				createMainNode(),
+				createNode(childType, { code: childCode }),
+			]);
+			await connectNodes(main, 'HAS_CURIOUS_CHILD', child);
+
+			const { status } = await patchHandler()(
+				getInput(
+					{
+						someString: 'some string',
+						curiousChild: {
+							code: childCode,
+							someString: null,
+						},
+					},
+					{ relationshipAction: 'replace', upsert: true },
+				),
+			);
+
+			expect(status).toBe(200);
+			expect(emitSpy).toHaveBeenCalledTimes(1);
+			expectUpdateEvent(mainType, mainCode, ['someString']);
```

Comment:
this helper is basically saying `expect(emitSpy).calledWith({type: mainType, code: mainCode, updatedProperties: ['someString']})`

Is it too DRY to have a helper for this?? I dunno. Before writing helpers like this and the one I mention a few comments above the tests were equally unreadable due to the masses of repeated, verbose object literals. At least with the helpers once you've learned them the tests are easier to scan. But open to discussion. Perhaps what's needed is some 'before you read the tests, know this...' documentation

Reply:
It's this bit I don't understand - not because it's been extracted into a helper but because it's not clear why the input (below) would result in this outcome. Are there separate tests for the diff algorithm?

```
{
    someString: 'some string',
    curiousChild: {
        code: childCode,
        someString: null,
    },
}
```